cvat-cli/src/cvat_cli/parser.py

```diff
@@ -369,6 +369,21 @@ def make_cmdline_parser() -> argparse.ArgumentParser:
         help="time interval between checks if archive processing was finished, in seconds",
     )
 
+    #######################################################################
+    # Auto-annotate
+    #######################################################################
+    auto_annotate_task_parser = task_subparser.add_parser(
+        "auto-annotate", description="Automatically annotate a CVAT task."
+    )
+    auto_annotate_task_parser.add_argument("task_id", type=int, help="task ID")
+    auto_annotate_task_parser.add_argument(
+        "--function",
+        help="name of module to use as the function",
+        metavar='MODULE',
+        required=True,
+        dest="function_module",
+    )
```

Comment:
Need to discuss this. From one point of view, some of these parameters can be convenient:
- job_id(s) - probably we should be able to annotate jobs as well. I agree. But I don't think that we need the argument here
- clear_existing - better to have just a separate command to remove annotation. Otherwise we can get very complicated interface.
- script file path - OK, instead of a module name
- strict - agree and maybe it should be a default mode. Thus we avoid many complains, that a function doesn't annotate a task. I met the problem today.
- skip-errors - I'm not sure we want to skip any errors.

Reply:
My thoughts:

> `job_id(s)`

I think ideally the CLI should have separate sets of task and job based commands (e.g. `cvat-cli task list` / `cvat-cli job list`). IIUC, job auto-annotation won't require you to use any task-based APIs, so it would make more sense to implement it as a job subcommand (`cvat-cli job auto-annotate`).

However, that would be beyond the scope of this PR.

> `clear_existing` / `append`

I agree with Maxim that this should be an option. Moreover, the default should probably be "append". Currently, the command clears existing annotations, but that's mainly just because it's easier. Defaulting to `append` will avoid accidental data loss.

@nmanovic I think we _should_ have a separate command to clear annotations, but it's not very convenient to run a separate command every time you want to reannotate, so I think an option for this command would be useful to have.

> script file path

I'm thinking of adding a `--function-file` option as an alternative to `--function` (and maybe renaming the latter to `--function-module`). Instead of importing, this will load the file as a string and compile it. It will require that the file has no relative imports, though.

> an extra parameter to for module root directory

I don't know what this means.

> `strict`

Yeah, fair point. I also agree with Nikita that this should be the default.

> `skip-errors`

I get the idea, but then what do you do with frames that were skipped? There's no easy way to find them afterwards, and even if we could find them, we can't rerun the function on just those frames.

We could consider some solution for this, but IMO it goes beyond the current scope.

Reply:
> Instead of importing, this will load the file as a string and compile it. It will require that the file has no relative imports, though.

> > an extra parameter to for module root directory

> I don't know what this means.

I've given an example in the test comments. Basically, it's needed both for relative imports and for fully-qualified imports to specify where to look for the import path (like `PYTHONPATH` does). Without this, it's hard to use scripts relying on a customized framework (e.g. mmdetection), lying somewhere near the cvat adapter script, or just outside the current directory:

```bash
# .../some/dir/mymodel/mmdetection/...
# .../some/dir/mymodel/cvat_adapter.py

cvat-cli auto-annotate --module-root '.../some/dir/' --module 'mymodel.cvat_adapter' ...
```

> > skip-errors

> I get the idea, but then what do you do with frames that were skipped? There's no easy way to find them afterwards, and even if we could find them, we can't rerun the function on just those frames.

I thought about things like OOM errors on some frames. Not sure how often it happens in real applications.

Actually, the implementation allows execution on specific frames (except for tracks), as the function is executed on separate frames, so it's only a question of API/CLI. I also see an option to execute the model only on some part of the task a useful feature, though the CLI use may be complicated in this case. Probably, we can leave this for future updates.

Reply:
> I've given an example in the test comments. Basically, it's needed both for relative imports and for fully-qualified imports to specify where to look for the import path (like PYTHONPATH does). 

So why not just use `PYTHONPATH`?

> Actually, the implementation allows execution on specific frames (except for tracks), as the function is executed on separate frames, so it's only a question of API/CLI. 

It's true that we can run the function on specific frames, but how do we know which frames to use (in the context of recovery from a partially successful auto-annotation run?

> I also see an option to execute the model only on some part of the task a useful feature, though the CLI use may be complicated in this case. Probably, we can leave this for future updates.

I agree.

Reply:
> So why not just use PYTHONPATH?

Probably, it's an implementation detail, and this obliges users to learn environment variables.

> how do we know which frames to use (in the context of recovery from a partially successful auto-annotation run?

It could be reported somehow, e.g. in the logs.

Reply:
I added `--clear-existing` and `--allow-unmatched-labels`. Correspondingly, I changed the defaults to not erase old annotations, and abort when a function declares labels not configured in the task.

> Probably, it's an implementation detail, and this obliges users to learn environment variables.

The user would have to learn something either way, and at least they might already know about `PYTHONPATH`.